### ISSUE TITLE
[BitSail][Optimization] Improve FakeSource connector.

### ISSUE THREAD

**Issue by BlockLiu** [bytedance/bitsail#243]
# Describe the problem or proposal

Improve FakeSource connectors.

# Task List

 - [ ] #242
 - [ ] #244  

# Are you willing to submit PR?
 - [ ] Yes I am willing to submit a PR.

**Comment by kyle-hawk** [comment]
Fake Connector does not seem to be able to declare whether the field is nullable, this can be useful sometimes 
@BlockLiu

**Comment by kyle-hawk** [comment]
Fake Connector support bigdecimal field, but seems not support specifying precision and scale, like this
```
        {
          "name": "decimal_type",
          "type": "bigdecimal(9,9)"
        }
```
this can be useful when we test decimal fields of sink connectors

**Comment by BlockLiu** [comment]
Yep, Fake connector is quite simple and can only be used for functional tests. It's weak for data verification tests.
Would you like to improve it?

**Comment by kyle-hawk** [comment]
sure, I'd like to join to improve this
